File: pytools/examples/lua/server/protocols/MsgTestPhp.lua
module("MsgTestPhp", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = MsgTestPhp})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.u16 = 0
end


function encode(self)
	pack = packet.new()
	pack:writeU16(self.u16)
	return pack:readBytes()
end

function decode(self, pack)
	self.u16 = pack:readU16()
	return self
end

function getBytes(self)
	return self:encode()
end


function setU16(self, u16)
	self.u16 = u16
end
function getU16(self)
	return self.u16
end

File: pytools/examples/golang/src/client/proto/msg_test_php.go
package proto

import (
	"packet"
)

type MsgTestPhp struct {
	u16                      uint16
}

func (this *MsgTestPhp) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint16(this.u16)

	return pack.ReadBytes()
}

func MsgTestPhpDecode(pack *packet.Packet) *MsgTestPhp {
	msgTestPhp := &MsgTestPhp{}

	msgTestPhp.u16 = pack.ReadUint16()
	return msgTestPhp
}

func (this *MsgTestPhp) SetU16(u16 uint16) {
	this.u16 = u16
}

func (this *MsgTestPhp) GetU16() uint16 {
	return this.u16
}
